src/components/HeaderSearch/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect, createRef } from 'react'
+import PropTypes from 'prop-types'
+import { Input, Icon, AutoComplete } from 'antd'
+import classNames from 'classnames'
+import Debounce from 'lodash-decorators/debounce'
+import Bind from 'lodash-decorators/bind'
+
+const styles = require('./index.less')
+
+// TODO: define on next task
+// static propTypes = {
+//   className: PropTypes.string,
+//   placeholder: PropTypes.string,
+//   onSearch: PropTypes.func,
+//   onPressEnter: PropTypes.func,
+//   defaultActiveFirstOption: PropTypes.bool,
+//   dataSource: PropTypes.array,
+//   defaultOpen: PropTypes.bool,
+//   onVisibleChange: PropTypes.func
+// }
+
+// static defaultProps = {
+//   defaultActiveFirstOption: false,
+//   onPressEnter: () => {},
+//   onSearch: () => {},
+//   className: '',
+//   placeholder: '',
+//   dataSource: [],
+//   defaultOpen: false,
+//   onVisibleChange: () => {}
+// }
+
+export default function HeaderSearch (props) {
+  const [searchMode, setSearchMode] = useState(props.defaultOpen)
+  const [value, setValue] = useState('')
+  // TO Verify: onVisibleChange not defined or passed as props
+  const { className, placeholder, open, onChange, onPressEnter, ...restProps } = props
+  let textInput = React.createRef()
+
+  let timeout
+
+  useEffect(() => {
+    if (searchMode) {
+      if (textInput.current) {
+        textInput.current.focus()
+      }
+    }
+    return function cleanup () {
+      clearTimeout(timeout)
+    }
+  }, [searchMode])
+
+  function onKeyDown (e) {
+    if (e.key === 'Enter') {
+      timeout = setTimeout(() => {
+        onPressEnter(value) // Fix duplicate onPressEnter
+      }, 0)
+    }
+  }
+
+  function onChangeInput (value) {
+    setValue(value)
+    if (onChange) {
```

Comment:
I would prefer to use `props.onChange` here, otherwise it's hard to know to which handler we are referring here.

Reply:
- [ ] use `props` here

Reply:
It is using the `onChange` from the props here.